<launch>
<!--
  <node type="pcl_pcdReadPublisher" pkg="beginner_tutorials" name="pcl_pcdReadPublisher" output="screen">
  </node>
  -->
  <!--
  <node type="pcl_voxel" pkg="beginner_tutorials" name="pcl_voxel" output="screen"/>
  -->
  <!--
  <node type="pcl_extractIndices" pkg="beginner_tutorials" name="pcl_extractIndices" output="screen">
    <remap from="points2" to="cloudVoxelFiltered"/>
  </node>
  -->
  <include file="$(find beginner_tutorials)/launch/play.launch"/>
  <node type="pcl_passThrough" pkg="beginner_tutorials" name="pcl_passThroughFilter" output="screen">
    <remap from="points2" to="camera/depth_registered/points"/>
    <remap from="output" to="cloudPassThroughFiltered"/>
  </node>

  <node type="pcl_voxel" pkg="beginner_tutorials" name="pcl_voxel" output="screen">
    <remap from="camera/depth_registered/points" to="cloudPassThroughFiltered"/>
  </node>
  <node type="pcl_statistical" pkg="beginner_tutorials" name="pcl_statistical" output="screen">
    <remap from="camera/depth_registered/points" to="cloudVoxelFiltered"/>
  </node>

  <!--node type="pcl_extractor_node" pkg="beginner_tutorials" name="pcl_extractor" output="screen">
    <remap from="points2" to="cloudVoxelFiltered"/>
    <remap from="output" to="extractIndices"/>
  </node-->
 
  <!--nodelet manager-->
  <node pkg="nodelet" type="nodelet" args="manager" name="extractor_manager" output="screen"/>
  <node pkg="nodelet" type="nodelet" name="pcl_extractor" args="load beginner_tutorials/PclExtractorNodelet extractor_manager --no-bond">
    <remap from="points2" to="cloudVoxelFiltered"/>
    <remap from="output" to="extractIndices"/>
    <rosparam file="$(find beginner_tutorials)/launch/params/kinectExtractorParams.yaml" command="load"/>
  </node>
 
  <!-- pkg="nodelet" type="nodelet" name="metric_rect" args="load depth_image_proc/convert_metric record_player_manager -no-bond"-->
  <node type="rviz" pkg="rviz" name="rviz" args="-d $(find beginner_tutorials)/launch/kinect.vcg"/>
  <node type="reconfigure_gui" pkg="dynamic_reconfigure" name="dynamic_reconfigure" />
  </launch>
